records_mover/db/redshift/records_copy.py

```diff
@@ -3,92 +3,121 @@
 from records_mover.records.records_format import (
     BaseRecordsFormat, DelimitedRecordsFormat, AvroRecordsFormat
 )
+from records_mover.records.delimited import ValidatedRecordsHints
 from records_mover.mover_types import _assert_never
 from sqlalchemy_redshift.commands import Format, Encoding, Compression
 from typing import Dict, Optional, Set
 
 RedshiftCopyOptions = Dict[str, Optional[object]]
 
 
-def redshift_copy_options(unhandled_hints: Set[str],
-                          records_format: BaseRecordsFormat,
-                          fail_if_cant_handle_hint: bool,
-                          fail_if_row_invalid: bool,
-                          max_failure_rows: Optional[int]) -> RedshiftCopyOptions:
-    redshift_options: RedshiftCopyOptions = {}
+class RedshiftCopyParamaters:
+    def __init__(self,
+                 records_format: DelimitedRecordsFormat,
+                 fail_if_cant_handle_hint: bool,
+                 unhandled_hints: Set[str],
+                 fail_if_row_invalid: bool,
+                 max_failure_rows: Optional[int]):
+        self.redshift_options: RedshiftCopyOptions = dict()
```

Comment:
having both `RedshiftCopyParameters` and `RedshiftCopyOptions` is kinda confusing to me

Reply:
refactored